### PR TITLE
build: replace gofmt with goimports

### PR DESCRIPTION
**Pull request by anjannath**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -247,8 +247,8 @@ e2e-story-registry: install
 
 
 .PHONY: fmt
-fmt:
-	@gofmt -l -w $(SOURCE_DIRS)
+fmt: $(TOOLS_BINDIR)/goimports
+	@goimports -l -w $(SOURCE_DIRS)
```
Comment (cfergeau) [inline Makefile]: needs to be `$(TOOLS_BINDIR)/goimports`

File: tools/tools.mk [hunk 2]
```diff
@@ -11,3 +11,6 @@ $(TOOLS_BINDIR)/gomod2rpmdeps: $(TOOLS_DIR)/go.mod
 
 $(TOOLS_BINDIR)/mockery: $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR) && GOBIN="$(TOOLS_BINDIR)" go install github.com/vektra/mockery/v2@latest
+
+$(TOOLS_BINDIR)/goimports: $(TOOLS_DIR)/go.mod
+	cd $(TOOLS_DIR) && GOBIN="$(TOOLS_BINDIR)" go install golang.org/x/tools/cmd/goimports@latest
```
Comment (cfergeau) [inline tools/tools.mk:16]: you need to add an import in https://github.com/crc-org/crc/blob/main/tools/dummy.go and update vendoring

**Review by cfergeau — COMMENTED** [review]
A few things to fix, but looks good.

File: tools/tools.mk (hunk 2 above)
Comment (anjannath) [reply tools/tools.mk:16]: ahh yes, updated now, not very clear this needs to be done, maybe we should add another target to the `tools.mk` to run `go mod tidy and go mod vendor`

Comment (cfergeau) [reply tools/tools.mk:16]: Without the addition to `dummy.go`, `go mod tidy && go mod vendor` will not vendor the newly added code :-/ We could check that `golang.org/x/tools/cmd/goimports` is in `dummy.go` somehow, but not worth spending a lot  of time on this.

**Review by cfergeau — APPROVED** [review]

**Pull request merged**